### PR TITLE
Allow non-string keys in get_treatment

### PR DESCRIPTION
**Pull request by maksimf**

### CODE REVIEW

File: lib/splitclient-rb/clients/split_client.rb [hunk 1]
```diff
@@ -107,9 +107,9 @@ def get_treatment(key, split_name, attributes = nil, split_data = nil, store_imp
     def keys_from_key(key)
       case key.class.to_s
       when 'Hash'
-        key.values_at(:bucketing_key, :matching_key).map(&:to_s)
-      when 'String'
-        [nil, key.to_s]
+        key.values_at(:bucketing_key, :matching_key).map { |k| k.nil? ? nil : k.to_s }
+      else
+        [nil, key].map { |k| k.nil? ? nil : k.to_s }
```
Comment (patricioe) [inline lib/splitclient-rb/clients/split_client.rb]: will String: "1" and Int: 1 hash to the same hash value?

**Comment by patricioe** [comment]
conflicts too

File: lib/splitclient-rb/clients/split_client.rb (hunk 1 above)
Comment (maksimf) [inline lib/splitclient-rb/clients/split_client.rb]: Yep, is this what's expected?

**Comment by maksimf** [comment]
@patricioe Fixed conflicts

File: lib/splitclient-rb/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module SplitIoClient
-  VERSION = '3.2.3'
+  VERSION = '3.2.4'
```
Comment (patricioe) [inline lib/splitclient-rb/version.rb]: @maksimf max please drop this change from here so it does not override of conflict with latest head.

File: lib/splitclient-rb/clients/split_client.rb [hunk 3]
```diff
@@ -107,9 +107,9 @@ def get_treatment(key, split_name, attributes = nil, split_data = nil, store_imp
     def keys_from_key(key)
       case key.class.to_s
       when 'Hash'
-        key.values_at(:bucketing_key, :matching_key)
-      when 'String'
-        [nil, key]
+        key.values_at(:bucketing_key, :matching_key).map { |k| k.nil? ? nil : k.to_s }
+      else
+        [nil, key].map { |k| k.nil? ? nil : k.to_s }
```
Comment (patricioe) [inline lib/splitclient-rb/clients/split_client.rb]: so

[number].to_s and "string".to_s have the same output? just checking.

**Comment by maksimf** [comment]
@patricioe Yes, same

**Comment by maksimf** [comment]
@patricioe conflicts fixed

File: lib/splitclient-rb/version.rb [hunk 4]
```diff
@@ -1,3 +1,3 @@
 module SplitIoClient
-  VERSION = '3.2.4-rc2'
+  VERSION = '3.2.5'
```
Comment (patricioe) [inline lib/splitclient-rb/version.rb]: please drop this. Let's change version only during releases.

**Comment by maksimf** [comment]
@patricioe got it, done

**Comment by patricioe** [comment]
merged thanks

**Pull request merged**